nodejs/aws-infra/iam/index.ts

```diff
@@ -0,0 +1,213 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "./../utils";
+
+export class User extends pulumi.ComponentResource {
+    public readonly user: aws.iam.User;
+    public readonly groupMembership?: aws.iam.UserGroupMembership;
+    public readonly accessKey?: aws.iam.AccessKey;
+
+    constructor(name: string, args: UserArgs = {}, opts?: pulumi.CustomResourceOptions) {
+        super(`awsinfra:x:iam:User`, name, args, opts);
+
+        // Save group membership and access key args so we can define them inline.
+        const groupMembership = args.groupMembership;
+        const accessKey = args.accessKey;
+
+        // Explicitly delete these props so we do *not* pass them into the User created
+        // below.
+        delete args.groupMembership;
+        delete args.accessKey;
+        delete args.user;
+
+        this.user = args.user || new aws.iam.User(name, args && args, { parent: this });
+
+        if (groupMembership !== undefined) {
+            this.groupMembership = new aws.iam.UserGroupMembership(
+                name,
+                { ...groupMembership, user: this.user.name },
+                { parent: this }
+            );
+        }
+
+        if (accessKey == "create") {
```

Comment:
intead of this string we can define things as such:

1. if args.accessKey is undefined, do nothing.
2. if it eixsts, we create an accesskey from whatever it is.

I'm not sure i see the need for an actual 'create'.

Reply:
I think this makes things much less clear, actually, because without `user`, none of the fields are required. So passing `accessKey: {}` would result in a creation of an access key, which seems super weird.

That said, I think it's fine to pull this out of the constructor, and put it into a method instead.